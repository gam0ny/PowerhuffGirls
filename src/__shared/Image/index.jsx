import { useState } from 'react';
import PropTypes from 'prop-types';

import classNames from 'class-names';
import noImageUrl from '../../styles/images/default-image.jpg';

const Image = ({className, src, width, height}) => {
    const [isLoaded, setIsLoaded] = useState(false);
    const [loadFailed, setLoadFailed] = useState(false);

    const figureClassNames = classNames(
        'image-wrapper',
        { 'image-wrapper--loading': !isLoaded && !loadFailed},
        className);

    const imageClassNames = classNames(
        'image-wrapper__image',
        { 'image-wrapper__image--hidden': !isLoaded }
    );

    return (
        <figure className={figureClassNames}
                style={{minWidth: width, minHeight: height}}>
            {loadFailed && <img className='image-wrapper__default-image' src={ noImageUrl }/>}
            <img className={imageClassNames}
                 src={src}
                 onLoad={ () => setIsLoaded(true) }
                 onError={ () => { setIsLoaded(false); setLoadFailed(true)} }
            />
        </figure>)
};

Image.propTypes = {
    className: PropTypes.string,
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
}

export default Image;