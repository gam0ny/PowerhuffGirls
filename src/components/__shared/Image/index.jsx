import {useState} from 'react';
import PropTypes from 'prop-types';

import classNames from 'class-names';
import noImageUrl from '../../../styles/images/default-image.jpg';

const Image = ({className, src, width, height}) => {
    const [isLoaded, setIsLoaded] = useState(false);
    const [loadFailed, setLoadFailed] = useState(false);

    const figureClassNames = classNames(
        'image-wrapper',
        {'image-wrapper--loading': !isLoaded && !loadFailed},
        className);

    const imageClassNames = classNames(
        'image-wrapper__image',
        {'image-wrapper__image--hidden': !isLoaded}
    );

    let component = null;

    if(src) {
        component =
          <figure className={figureClassNames}
                    style={{minWidth: width, maxWidth: width, minHeight: height, maxHeight: height}}>
              {loadFailed && <img className='image-wrapper__default-image' src={noImageUrl} alt={'default'}/>}
              <img className={imageClassNames}
                   src={src}
                   alt={'target'}
                   onLoad={() => setIsLoaded(true)}
                   onError={() => {
                     setIsLoaded(false);
                     setLoadFailed(true)
                   }}
              />
          </figure>;
    }

    return component;
};

Image.propTypes = {
    className: PropTypes.string,
    src: PropTypes.string,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
}

export default Image;
