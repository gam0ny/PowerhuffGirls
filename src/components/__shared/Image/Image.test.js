import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Image from "./index";

describe('[Image] component test', () => {
    it('[Image] component renders properly', () => {
        const nextProps = {
            src: 'url',
            width: 100,
            height: 100,
        };

        const imageComponent = shallow(<Image {...nextProps}/>);

        expect(toJson(imageComponent)).toMatchSnapshot();
    });
})
