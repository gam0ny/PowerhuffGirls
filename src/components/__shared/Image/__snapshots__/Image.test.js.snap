// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Image] component test [Image] component renders properly 1`] = `
<figure
  className="image-wrapper image-wrapper--loading"
  style={
    Object {
      "maxHeight": 100,
      "maxWidth": 100,
      "minHeight": 100,
      "minWidth": 100,
    }
  }
>
  <img
    alt="target image"
    className="image-wrapper__image image-wrapper__image--hidden"
    onError={[Function]}
    onLoad={[Function]}
    src="url"
  />
</figure>
`;
